core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryBuilder.java

```diff
@@ -19,115 +19,299 @@
-        this.ignoreMalformed = ignoreMalformed;
+        if (coerce == false) {
+            this.ignoreMalformed = ignoreMalformed;
+        }
         return this;
     }
 
     @Override
+    protected Query doToQuery(QueryShardContext shardContext) throws IOException {
+        QueryValidationException exception = validate(shardContext.indexVersionCreated().before(Version.V_2_0_0));
+        if (exception != null) {
+            throw new QueryShardException(shardContext, "couldn't validate latitude/ longitude values", exception);
+        }
+
+        if (coerce) {
+            GeoUtils.normalizePoint(center, coerce, coerce);
+        }
+
+        double normDistance = geoDistance.normalize(this.distance, DistanceUnit.DEFAULT);
+
+        MappedFieldType fieldType = shardContext.fieldMapper(fieldName);
+        if (fieldType == null) {
+            throw new QueryShardException(shardContext, "failed to find geo_point field [" + fieldName + "]");
+        }
+        if (!(fieldType instanceof GeoPointFieldMapper.GeoPointFieldType)) {
+            throw new QueryShardException(shardContext, "field [" + fieldName + "] is not a geo_point field");
+        }
+        GeoPointFieldMapper.GeoPointFieldType geoFieldType = ((GeoPointFieldMapper.GeoPointFieldType) fieldType);
+
+        IndexGeoPointFieldData indexFieldData = shardContext.getForField(fieldType);
+        Query query = new GeoDistanceRangeQuery(center, null, normDistance, true, false, geoDistance, geoFieldType, indexFieldData, optimizeBbox);
+        return query;
+    }
+
+    @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(NAME);
-        if (geohash != null) {
-            builder.field(name, geohash);
-        } else {
-            builder.startArray(name).value(lon).value(lat).endArray();
-        }
+        builder.startArray(fieldName).value(center.lon()).value(center.lat()).endArray();
         builder.field("distance", distance);
         if (geoDistance != null) {
             builder.field("distance_type", geoDistance.name().toLowerCase(Locale.ROOT));
         }
-        if (optimizeBbox != null) {
-            builder.field("optimize_bbox", optimizeBbox);
+        builder.field("optimize_bbox", optimizeBbox);
+        builder.field("coerce", coerce);
+        builder.field("ignore_malformed", ignoreMalformed);
+        printBoostAndQueryName(builder);
+        builder.endObject();
+    }
+
+    @Override
+    public int doHashCode() {
+        return Objects.hash(center, geoDistance, optimizeBbox, distance, coerce, ignoreMalformed);
+    }
+
+    @Override
+    public boolean doEquals(GeoDistanceQueryBuilder other) {
+        return Objects.equals(this.fieldName, other.fieldName) &&
+                (this.distance == other.distance) &&
+                (this.coerce == other.coerce) &&
+                (this.ignoreMalformed == other.ignoreMalformed) &&
```

Comment:
nit: remove `this` as its not needed and will be consistent with the below lines